include/assertion.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2022 Felix Gohla, Konrad Hanff, Tobias Kantusch,
+ *                    Quentin Kuth, Felix Roth. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+#pragma once
+
+#include "dev.h"
+#include "largeblob.h"
+#include "sha256.h" // TODO: Use Monocypher with abstraction
+
+// ed25519 signatures are 512 bits long
+// We do not support other (longer) signatures for now.
+#define ASSERTION_SIGNATURE_LENGTH 64
+
+// The standard says 1023, see https://github.com/w3c/webauthn/pull/1664.
+// see https://github.com/solokeys/fido-authenticator/pull/8
+// We define this as our maximum in order to reduce stack usage.
+#define ASSERTION_MAX_KEY_HANDLE_LENGTH 255
+
+// This is not defined by the standard.
+// However, we define this as our limit.
+#define ASSERTION_AUTH_DATA_LENGTH 128
+#define ASSERTION_AUTH_DATA_RPID_HASH_LEN 32
+
+// A SHA256 hash.
+#define ASSERTION_CLIENT_DATA_HASH_LEN 32
+
+#define ASSERTION_PRE_IMAGE_LENGTH ASSERTION_AUTH_DATA_LENGTH + SHA256_DIGEST_LENGTH
+
+typedef void es256_pk_t;
+
+typedef struct fido_assert_blob {
+    uint8_t const  *ptr;
+    size_t          len;
+} fido_assert_blob_t;
+
+#define FIDO_ASSERT_EXTENSION_LARGE_BLOB_KEY        FIDO_EXT_LARGEBLOB_KEY
+typedef uint8_t fido_assert_ext_t;
+
+#define FIDO_ASSERT_OPTION_UP                       BITFIELD(0)
+#define FIDO_ASSERT_OPTION_UV                       BITFIELD(1)
+typedef uint8_t fido_assert_opt_t;
+
+#define FIDO_CREDENTIAL_TYPE_PUBLIC_KEY             BITFIELD(0)
+typedef uint8_t fido_cbor_credential_type_t;
+
+typedef struct fido_cbor_credential {
+    fido_cbor_credential_type_t type;               // credential type
+    uint8_t id[ASSERTION_MAX_KEY_HANDLE_LENGTH];    // credential id
+    uint8_t id_length;                              // The length of the credential id.
+} fido_cbor_credential_t;
+
+// User Presence
```

Comment:
To resolve future confusion by others similar to [my comment](https://github.com/All-Your-Locks-Are-Belong-To-Us/libmicrofido2/pull/19/files#r925537584).

```suggestion
// Authenticator Data as defined in https://www.w3.org/TR/webauthn-2/#authenticator-data
// User Presence
```

Reply:
The same link is already below this one. So you want to move it up?

Reply:
Moved the comment. :)